nixos/tests/kernel-generic.nix

```diff
@@ -7,6 +7,89 @@
 with pkgs.lib;
 
 let
+  patchedPkgs = pkgs.extend (
+    final: prev: {
+      linuxKernel = prev.linuxKernel // {
+        kernelPackagesExtensions = prev.linuxKernel.kernelPackagesExtensions ++ [
+          (
+            finalKernelPackage: _:
+            let
+              inherit (finalKernelPackage) kernel;
+            in
+            {
+              hello-world = final.stdenv.mkDerivation {
+                name = "hello-module";
+
+                nativeBuildInputs = kernel.moduleBuildDependencies;
+                makeFlags = kernel.commonMakeFlags ++ [
+                  # Variable refers to the local Makefile.
+                  "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+                  # Variable of the Linux src tree's main Makefile.
+                  "INSTALL_MOD_PATH=$(out)"
+                ];
+
+                buildFlags = [ "modules" ];
+                installTargets = [ "modules_install" ];
+
+                src = final.linkFarm "hello-module-src" [
+                  rec {
+                    name = "hello.c";
+                    path = final.writeTextFile {
+                      inherit name;
+                      text = ''
+                        #include <linux/module.h>
+                        #include <linux/printk.h>
+
+                        static int hello(void)
+                        {
+                            pr_info("Hello world!");
+                            return 0;
+                        }
+                        module_init(hello);
+
+                        static void goodbye(void)
+                        {
+                            pr_info("Goodbye");
+                        }
+                        module_exit(goodbye);
+
+                        MODULE_LICENSE("GPL");
```

Comment:
Presumably we're not using any GPL symbols?  Ideally we don't have GPL code inside MIT-licensed Nix files…